enhancements/monitoring/hypershift-monitoring.md

```diff
@@ -0,0 +1,392 @@
+  
+> Exact list of what is scraped on guest clusters for HSRE does not matter. All we need to know: some service monitors.
+
+* HSRE can query metrics data over SLO period (usually 28 days).
+  * Longer durations nice to have, but technically not required. There is also option that allows HSRE to persist SLO from platform to [different backend if needed](https://docs.google.com/document/d/1BJarERppgiJ8esc6d8anbJMOQ0AflFBQea-Zc9wAp0s/edit?disco=AAAAQTFsbwM).
+* HSRE have an easy way to monitor dozens of HyperShift clusters in multiple geo regions and thousands of control planes in total. This means in particular:
+  * Ability to perform PromQL across fleet metrics
+
+> NOTE(Mariusz): Alerts can be region wide only, and query in different tab per region might be good enough for now.
+
+  * Ability to see logs across the fleet to see a pattern
+  * No need to tunnel to specific control plane to query metrics or see logs
+  * Centralized alerting
+* CUS can monitor / use HPA/VPA / vis with local console using metrics FROM their workloads on guest clusters or from nodes on guest clusters
+  * Container metrics (about their containers).
+  * This includes part of Kube State Metrics (Kubernetes metrics) that relate to their workloads.
+* BU can gather required telemetry using the Telemeter pipeline.
+
+### Non-Goals
+
+* Solve Logging and Tracing needs.
+* Provide scalable storage solution for customer metrics. We just want to ensure UWM, their DIY,
+console or vendor monitoring does not have major blockers when integrating with this
+architecture.
+* Describe Managed Services (Addons) monitoring in detail. We assume Managed Services are just “Customer workloads”, where the MTSRE is just a customer. See [What About Addons FAQ](#what-about-layered-service--addons-monitoring) for more details.
+* Solve observability challenges brought by KCP. KCP has different trade-offs.
+* Provide any other UI than Grafana for our Managed HSRE needs.
+* Provide any other querying APIs than Prometheus based on PromQL
+* Give customers access to any control plane metrics unrelated to their workloads (this is similar in GKE).
+* Provide a solution for non-HyperShift OSD. In fact, the solution for OSD observability is planned to be really similar to HyperShift. 
+This is because OSD observability struggles on similar points--inability to aggregate observability data in a cheap way on a global view.
+However, since HyperShift has no solution at all, for now starting from HyperShift sounds like a great opportunity.
+
+## Proposal
+
+### Personas
+
+* HyperShift Admin SRE (HSRE): People that will operate the Hypershift management cluster and its control planes. In practice, this means the SD org on our managed solutions.
+* Customers (CUS) People that attach data planes to HyperShift and run workloads on data planes.
+* Business Unit (BU): Red Hat business unit who is interested in telemetry and data from OpenShift deployments.
+
+### User Stories
+
+TBD
```

Comment:
This has to be fleshed out before this enhancement should be merged.

Reply:
Do you mind helping us with that? @jeremyeder @csrwng